### ISSUE TITLE
WebRTC still leak my local IP

### ISSUE THREAD

**Issue by Nottt** [gorhill/uBlock#542]
I have disabled the "leakage" option for testing purposes , then after reactivate it, https://diafygi.github.io/webrtc-ips/ still reveal my local IP. Happened on Chrome 44.0.2403.125 m (64-bit) on Windows 7 x64. With latest Ublock Origin version.

I tried also turning off and on. To reproduce this just disable this option and enable it again and see if your IP leak.

Edit : It's appear to be working now, but only after i restarted my PC. Probably a bug on Chrome handling the API call ?

**Comment by gjallarhorn** [comment]
Please follow this guide before submitting a new issue:
https://github.com/gorhill/uBlock/blob/master/CONTRIBUTING.md

I can't reproduce this on my Windows 7 machine, running Firefox 39 and uBlock Origin 1.0.0.0

**Comment by gorhill** [comment]
Can't reproduce with Chromium 43, Linux Mint 17, uBlock Origin 1.0.0.1.

**Comment by gorhill** [comment]
Marking as "invalid" because not even the most basic information is provided, like the browser/version.

**Comment by CODYQX4** [comment]
For the record, I just updated to Chrome 44.0.2403.130 on OS X 10.10.4, with the option enabled, and the LAN IP does leak. (uBlock 1.0.0.0)

I closed and reopened the browser, as well as toggled the option. Still leaks the LAN IP.

**Comment by gorhill** [comment]
> the LAN IP does leak

Is there anything reported in the dev console of uBlock Origin? (accessible from _Extensions_ page).

**Comment by gorhill** [comment]
I tried with Google Chrome 46.0.2467.2 dev (64-bit) on Linux, and can't reproduce. So I believe you should report this to Chromium devs, as it appears the issue is specific to Chrome 44.0.2403.130 on OS X 10.10.4, as I am not able to reproduce here. uBlock merely disable the relevant settings using the [chrome API](https://developer.chrome.com/extensions/privacy#property-network), so if it's not working, the issue lie with the browser.

**Comment by CODYQX4** [comment]
I made an extension for this before it was a uBlock feature:
https://github.com/CODYQX4/WebRTC-Multiple-Routes

That does, in fact, work for me, on the browser in question. On opening Chrome, ON means we aren't blocking, OFF means we're fixing the IP. That seems to work, uBlock seems to have no effect. If you click the button, I force a chrome://restart, as I had issues making it take effect immediately when I developed it. All testing was done on OS X (10.9 and 10.10) on Chrome 42, and I stopped using it when uBlock added the feature. I only noticed this missing today.

I read the value of the preference upon initial extension load. Presumably, toggling it in uBlock and then restarting Chrome should cause me to read the opposite value. Nothing seems to take effect.

Only console error I had was failing to connect to someonewhocares.org.

PS: Looks like uBlock was working on Chrome 44.0.2403.125, and still worked on Chrome 44.0.2403.130 on my other Mac (also 10.10.4). That one is using a VPN, this one in question is not.

**Comment by gorhill** [comment]
> I read the value of the preference upon initial extension load.

My understanding is that only one extension can control a setting. For example, in the following picture, it shows uBlock is controlling the pre-fetch setting:

![...](https://cloud.githubusercontent.com/assets/585534/7914528/924b9314-0845-11e5-8012-f67e4b1814cd.png)

Unfortunately, the WebRTC setting has no UI in Chrome, so we can't see what the picture above shows. Is there another extension controlling the WebRTC setting?

**Comment by CODYQX4** [comment]
There shouldn't be. Again, I had the issue without mine installed.

I have the same extensions on both machines.

**Comment by gorhill** [comment]
I don't know what else to say than to submit the issue to Chromium devs.

**Comment by Manouchehri** [comment]
I was able to reproduce the issue by setting a proxy with SwitchyOmega; I'm not sure if it's because uBlock is guessing the wrong IP (since both addresses are technically not local), or if it never even gets a chance to see the WebRTC request.

**Comment by gorhill** [comment]
> uBlock is guessing the wrong IP

uBlock does not guess any IP, it merely toggles a browser setting meant to prevent local IP address leaks: [`chrome.privacy.network.webRTCMultipleRoutesEnabled`](https://code.google.com/p/chromium/issues/detail?id=457629).

**Comment by Manouchehri** [comment]
Thanks for clarifying. Definitely a Chrome bug then.
